traversal/procedure/ProcedureEdge.java

```diff
@@ -362,7 +362,7 @@ static class Backward extends Isa<ProcedureVertex.Type, ProcedureVertex.Thing> {
                     if (to.props().hasIID()) {
                         return to.iterateAndFilterFromIID(graphMgr, params).filter(vertex -> isaTypes.contains(vertex.type()));
                     } else {
-                        FunctionalIterator<TypeVertex> toTypes = iterate(isaTypes).filter(v -> to.props().types().contains(type.properLabel()));
+                        FunctionalIterator<TypeVertex> toTypes = iterate(isaTypes).filter(v -> to.props().types().contains(v.properLabel()));
```

Comment:
In the test, the inferred types of `$x` only contain `person`. When `type` is `thing` or `entity`, the lambda body always returned false, even though `person` is in the set `isaTypes`.